subscription/update_my_subscription_data.go

```diff
@@ -0,0 +1,32 @@
+// Package subscription implements the Subscription NEX protocol
+package subscription
+
+import (
+	"fmt"
+
+	nex "github.com/PretendoNetwork/nex-go"
+	"github.com/PretendoNetwork/nex-protocols-go/globals"
+)
+
+// UpdateMySubscriptionData sets the UpdateMySubscriptionData handler function
+func (protocol *SubscriptionProtocol) UpdateMySubscriptionData(handler func(err error, client *nex.Client, callID uint32, unk uint32, content []byte)) {
+	protocol.updateMySubscriptionDataHandler = handler
+}
+
+func (protocol *SubscriptionProtocol) handleUpdateMySubscriptionData(packet nex.PacketInterface) {
+	if protocol.updateMySubscriptionDataHandler == nil {
+		fmt.Println("[Warning] SubscriptionProtocol::UpdateMySubscriptionData not implemented")
+		go globals.RespondError(packet, SubscriptionProtocolID, nex.Errors.Core.NotImplemented)
+		return
+	}
+
+	client := packet.Sender()
+	request := packet.RMCRequest()
+
+	callID := request.CallID()
+	parameters := request.Parameters()
+
+	//This is done since the server doesn't need to care about the data here (it's game-specific), so we just pass it along to store however the handler wants
+	content := parameters[4:]
+	go protocol.updateMySubscriptionDataHandler(nil, client, callID, 0, content)
```

Comment:
We should read the four bytes of the parameters, even if it looks like it's always 0

Reply:
This also goes for the other method with the cut 8 bytes

Reply:
> This also goes for the other method with the cut 8 bytes

The 8 bytes of the other method are being read? https://github.com/PretendoNetwork/nex-protocols-go/pull/33/files#diff-325c5c0c44b4892c2796757f60852782da1df50dc53810a41e629981e7faeebfR30

Reply:
My apologies I didn't see that portion

That being said, instead of just slicing like this it would be cleaner to use the `ReadBytesNext` method on the stream classes, which just takes in a length

I wonder if adding something like a `ReadRemaining` method would be appropriate in the stream class to cover these cases? I don't believe we have anything like that right now